src/helpers/binding.ts

```diff
@@ -0,0 +1,163 @@
+const placeholderRegExp =
+  /["'`](?:(?<=")[^"\\]*(?:\\.[^"\\]*)*"|(?<=')[^'\\]*(?:\\.[^'\\]*)*'|[^`]*`)|(\?\d*|[:@$][a-zA-Z_]\w+)/gmu;
+
+function isPlainObject(obj: any): obj is Record<string, BaseType> {
```

Comment:
It is almost 2023, and it is still hard to know if something is a plan ol' javascript object or not...